Source/LuaBridge/RefCountedObject.h

```diff
@@ -302,56 +299,12 @@ class RefCountedObjectPtr
     ReferenceCountedObjectClass* referencedObject;
 };
 
-/** Compares two ReferenceCountedObjectPointers. */
-template<class ReferenceCountedObjectClass>
-bool operator==(const RefCountedObjectPtr<ReferenceCountedObjectClass>& object1,
-                ReferenceCountedObjectClass* const object2)
-{
-    return object1.getObject() == object2;
-}
-
-/** Compares two ReferenceCountedObjectPointers. */
-template<class ReferenceCountedObjectClass>
-bool operator==(const RefCountedObjectPtr<ReferenceCountedObjectClass>& object1,
-                const RefCountedObjectPtr<ReferenceCountedObjectClass>& object2)
-{
-    return object1.getObject() == object2.getObject();
-}
-
-/** Compares two ReferenceCountedObjectPointers. */
-template<class ReferenceCountedObjectClass>
-bool operator==(ReferenceCountedObjectClass* object1,
-                RefCountedObjectPtr<ReferenceCountedObjectClass>& object2)
-{
-    return object1 == object2.getObject();
-}
-
-/** Compares two ReferenceCountedObjectPointers. */
-template<class ReferenceCountedObjectClass>
-bool operator!=(const RefCountedObjectPtr<ReferenceCountedObjectClass>& object1,
-                const ReferenceCountedObjectClass* object2)
-{
-    return object1.getObject() != object2;
-}
-
-/** Compares two ReferenceCountedObjectPointers. */
-template<class ReferenceCountedObjectClass>
-bool operator!=(const RefCountedObjectPtr<ReferenceCountedObjectClass>& object1,
-                RefCountedObjectPtr<ReferenceCountedObjectClass>& object2)
-{
-    return object1.getObject() != object2.getObject();
-}
-
-/** Compares two ReferenceCountedObjectPointers. */
-template<class ReferenceCountedObjectClass>
-bool operator!=(ReferenceCountedObjectClass* object1,
-                RefCountedObjectPtr<ReferenceCountedObjectClass>& object2)
-{
-    return object1 != object2.getObject();
-}
-
 //==============================================================================
 
+// forward declaration
+template<class T>
+struct ContainerTraits;
+
```

Comment:
Is this really required? The definition is right below.

Reply:
This forward declaration is required because the definition below is a template specialization of `ContainerTraits<T>` which has to be known before.

It's required because I've removed these includes:
- `LuaBridge/detail/Config.h`
- `LuaBridge/detail/TypeTraits.h`

The same thing was already done in `LuaBridge/RefCountedPtr.h`.

Reply:
Interesting is that this was already done nearly 10 years ago: 7f4e5dda14be39bb53d9b767ac9c8159effed015

Reply:
A forward declaration is required where there's no access to a real definition or a having a definition isn't possible.
Example for the 2nd case is there a class A refers to a class B and the B refers to the A. The cyclic dependency is broken with a B forward declaration prior to the class A definition and the subsequent class B definition.
Any class always has access to itself. E.g. the class A can refer the the class A anywhere.

Reply:
Indeed, I completely agree but the class definition below is still a template specialization of `ContainerTraits<T>` which need to be known before the specialization.

Anyway I've tried to compile the file without the forward declaration and it results in the following error.
![image](https://user-images.githubusercontent.com/83842777/146609184-825e9eb2-0a27-497f-b40d-291ee6e7cedd.png)

During this I've just recognized that the header for `std::swap` is missing. I'm going to open a pull request for that.